### DISCUSSION TITLE
Request getting http 500 from server when executed by Cypress

### DISCUSSION THREAD

**Discussion by MaxwellCravo** [cypress-io/cypress#26046] [Questions and Help]
Hi guys, I have a code that login in the application, fill some fields and then I submit it by clicking in a botton.
Is it a easy thing right? no, it doesn't, I'm trying since this morning submit this shit and I'm still getting this http 500 error.
It happen only when I'm using Cypress. If I do it manually, it works. If I use postman, it works...

Now lets go to the things I guess is the error.

When i'm using cypress I saw my request not sending the origin key in the header. its the only thing different that I found.

I change my chrome web security options to false
I tryed to use the cy.origin() function but it always fail with this message

`cy.origin() requires the first argument to be a different domain than top. You passed <myURL> to the origin command, while top is at <myURL>.`

I dont know what I can do anymore.

If I don't fix it, I will have to use Robot Framework to automate this project, and guys.... I HATE ROBOT FRAMEWORK.

please help me, how to set my cypress to send the origin on the requests?

**Comment by portyu9** [comment]
Hi @MaxwellCravo 

`cy.origin()` is not needed here—the error confirms you are already on the same origin.

The important part is `chromeWebSecurity: false`. In Chrome/Electron, disabling web security can cause the browser to stop sending the `Origin` header.

Remove it (or set it back to the default `true`) and fully restart Cypress:

```js
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  e2e: {
    chromeWebSecurity: true,
  },
})
```

If you absolutely need to override the header for a specific request, you can also do:

```js
cy.intercept('POST', '**/your-endpoint', (req) => {
  req.headers.origin = 'https://your-domain.com'
})
```

But the primary fix is to **remove `chromeWebSecurity: false` and restart Cypress**. `cy.origin()` is only for executing Cypress commands on a different origin.

**Reply by portyu9** [reply]
@MaxwellCravo Please mark this answer as accepted if it helped, thank you.
